bitcoin/script/miniscript.h

```diff
@@ -519,6 +519,7 @@ struct Node {
                     next_sats.push_back(sats[sats.size() - 1] + sub->ops.sat);
                     sats = std::move(next_sats);
                 }
+                assert(k < sats.size());
```

Comment:
In c303afd65f99fca22f70ef60b16b824f989d1029, all the assertions should be `k <= sats.size()` instead of `k < sats.size()` as it's now possible and that's what you are checking for in the previous commit.

Reply:
Indeed. I should have been more careful here!

Reply:
And i should have had tests to my PR updating the thresh bounds :) #59 fixes this.